File: src/test/java/dev/retrotv/crypt/owe/md/MD5Test.java
package dev.retrotv.crypt.owe.md;

import dev.retrotv.enums.Algorithm;
import dev.retrotv.crypt.owe.OWETest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class MD5Test extends OWETest {

    @Test
    @DisplayName("MD5 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(Algorithm.MD5);
    }

    @Test
    @DisplayName("MD5 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new MD5(), Algorithm.MD5);
    }

    @Test
    @DisplayName("MD5 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new MD5());
    }

    @Test
    @DisplayName("MD5 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new MD5());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/md/MD5Test.kt
package dev.retrotv.crypto.owe.hash.md

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class MD5Test : OWETest() {
    @Test
    @DisplayName("MD5 File hash 테스트")
    @Throws(Exception::class)
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.MD5)
    }

    @Test
    @DisplayName("MD5 File hash matches 테스트")
    @Throws(Exception::class)
    fun fileHashMatchesTest() {
        fileHashMatchesTest(MD5(), Algorithm.Hash.MD5)
    }

    @Test
    @DisplayName("MD5 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(MD5())
    }
}
